commands/share.js

```diff
@@ -0,0 +1,94 @@
+var chalk = require('chalk')
+var prettyBytes = require('pretty-bytes')
+var Dat = require('../lib/dat')
+var logger = require('../lib/logger')
+var ui = require('../lib/ui')
+
+module.exports = function (args) {
+  var dat = Dat(args)
+  var log = logger(args)
+
+  var addText = 'Adding '
+  var updated = false
+  var initFileCount = 0
+  var swarmTimeout = null
+
+  log.status('Starting Dat...\n', 0)
+  if (args.snapshot) log.status('Creating Link...', 1)
+  else log.status('Connecting...', 1)
+
+  dat.on('ready', function () {
+    log.message('Sharing ' + dat.dir + '\n')
+    dat.share(function (err) {
+      if (err) onerror(err)
+    })
+  })
+
+  dat.on('file-counted', function () {
+    var msg = 'Calculating Size: '
+    msg += dat.appendStats.files + ' files '
+    msg += chalk.dim('(' + prettyBytes(dat.appendStats.bytes) + ')')
+    log.status(msg + '\n', 0)
+  })
+
+  dat.on('key', function (key) {
+    log.message(ui.keyMsg(key))
+    if (args.quiet) console.log(ui.keyMsg(key))
+  })
+
+  dat.on('file-added', printStats)
+  dat.on('file-exists', printStats)
+
+  dat.on('append-ready', printStats)
+
+  dat.on('archive-finalized', function () {
+    addText = 'Added '
+    initFileCount = dat.stats.filesTotal
+    printStats()
+  })
+
+  dat.on('archive-updated', function () {
+    addText = 'Updated '
+    updated = true
+    printStats()
+  })
+
+  dat.on('connecting', function () {
+    var msg = 'Waiting for connections. '
+    if (dat.archive.live) msg += 'Watching for updates...'
+    log.status(msg, 1)
+  })
+
+  dat.on('swarm-update', printSwarm)
+  dat.on('upload', printSwarm)
+
+  function printSwarm () {
+    log.status(ui.swarmMsg(dat), 1)
+    if (swarmTimeout) clearInterval(swarmTimeout)
+    swarmTimeout = setInterval(function () {
+      log.status(ui.swarmMsg(dat), 1)
+    }, 500)
+  }
+
+  function printStats (data) {
+    var stats = dat.stats
+    var files = stats.filesTotal
```

Comment:
@karissa they overlap and could definitely use some improvement/more thought. One issue I keep running into is showing files vs "items" (files + dirs), since hyperdrive uses the combined total for it's count (`archive.metadata.blocks`).

`.appendStats` is files, dirs, and bytes calculated on sharing start. `.stats` is always the current archive stats, which gets updated as files are appended (initially and live).

```
stats = {
    filesTotal: 0,
    bytesTotal: 0,
    bytesUp: 0,
    bytesDown: 0,
    rateUp: speedometer(),
    rateDown: speedometer()
}

appendStats = {
    files: 0,
    bytes: 0,
    dirs: 0
}
```